contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,564 @@
+
+BACKGROUND: This provides a way to dynamically create raw block devices to later be consumed by software defined storage.
+
+WORKFLOW:
+
+ADMIN:
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: aws-ebs-raw
+provisioner: kubernetes.io/aws-ebs-raw
+parameters:
+  type: gp2
+  fstype: raw 
+```
+
+* Storage Class Stateful Set
+
+```
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 2
+  template:
+    metadata:
+      labels:
+	app: nginx
+    spec:
+      containers:
+      - name: nginx
+	image: gcr.io/google_containers/nginx-slim:0.8
+	ports:
+	- containerPort: 80
+	  name: web
+	volumeMounts:
+	- name: www
+	  mountPath: /usr/share/nginx/html
+ volumeClaimTemplates:
+  - metadata:
+      name: datadir
+      annotations:
+	volume.beta.kubernetes.io/storage-class: aws-ebs-raw
+    spec:
+      accessModes:
+	- "ReadWriteOnce"
+      resources:
+	requests:
+	  storage: 10Gi
+```
+***SUITABLE FOR: NETWORK ATTACHED BLOCK***
+
+## UC8: 
+
+DESCRIPTION: Admin creates network raw block devices
+
+BACKGROUND: Admin wishes to pre-create Persistent Volumes in GCE as raw block devices
+
+WORKFLOW:
+
+ADMIN:
+
+```
+apiVersion: "v1"
+kind: "PersistentVolume"
+metadata:
+  name: gce-disk-1
+  annotations:
+    volume.beta.kubernetes.io/mount-options: "discard"
+Spec:
+  volumeType: block
+  capacity:
+    storage: "10Gi"
+  accessModes:
+    - "ReadWriteOnce"
+  gcePersistentDisk:
+    fsType: "block"
+    pdName: "gce-disk-1"
+```
+
+***If admin specifies volumeType: block + fstype: ext4 then they would have the default behavior of files on block ***
+***fsType values will be provisioner dependent. Block is suggested for development simplicity. Since the PVC object is passed
+   to the provisioner, it will be responsible for validating and handling whether or not it supports the volumeType being passed ***
+
+# Container Runtime considerations
+It is important the values that are passed to the container runtimes are valid and support the current implementation of these various runtimes. Listed below are a table of various runtime and the mapping of their values to what is passed from the mount. 
+
+| runtime engine    | runtime options  | accessMode       |  
+| --------------    |:----------------:| ----------------:|
+| docker/runc/rkt   |  mknod / RWM     | RWO              |
+| docker/runc/rkt   |       R          | ROX              |
+
+The accessModes would be passed as part of the options array and would need validate against the specific runtime engine. 
+Since rkt doesn't use the CRI, the config values would need to be passed in the legacy method.
+
+The runtime option would be placed in the DeviceInfo as such:
+devices = append(devices, kubecontainer.DeviceInfo{PathOnHost: path, PathInContainer: path, Permissions: "mrw"}) for RWO
```

Comment:
Maybe this needs be explicitly stated in this proposal, but my understanding is that the Pod.volumeMounts.mountPath  is overloaded to work for block devices.  If the PVC.volumeType == block, then it will populate the CRI.DeviceInfo.  And if it is filesystem, then it will populate the VolumeInfo (like today).